### PR TITLE
Docs/2_4 upgrade

### PR DESCRIPTION
**Pull request by colmazia**

### CODE REVIEW

**Review by babybunn — APPROVED** [review]

**Comment by taobun** [comment]
Did we have format config? @babybunn

File: docs/joining-mainnet/join-as-a-validator-using-statesync.md [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+```
+
+### Step 1.4: Setup seeds or persistence peers of your node to the network.
+
+This can be done by editing `seeds` or `persistent_peers` property in `$HOME/.band/config/config.toml`.
+Please see [here](https://github.com/bandprotocol/launch/tree/master/laozi-mainnet) for the list of seeds and peers.
+
+```bash
+# List of seeds and persistent peers you want to add
+export SEEDS="<SEED>,<SEED>,..."
```
Comment (RogerKSI) [inline docs/joining-mainnet/join-as-a-validator-using-statesync.md]: please add example

**Review by RogerKSI — CHANGES_REQUESTED** [review]

File: docs/joining-mainnet/join-as-a-validator-using-statesync.md [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+source ~/.profile
+```
+
+Go binary should be at `/usr/local/go/bin` and any executable compiled by `go install` command should be at `~/go/bin`
+
+### Step 1.2: Clone & Install Bandchain Laozi
+
+```bash
+# Clone Bandchain Laozi version v2.4.1
+git clone https://github.com/bandprotocol/chain
```
Comment (RogerKSI) [inline docs/joining-mainnet/join-as-a-validator-using-statesync.md]: add cd ~

**Review by RogerKSI — CHANGES_REQUESTED** [review]

**Comment by babybunn** [comment]
> Did we have format config? @babybunn

I already added the formatter config to the branch @colmazia please check

File: docs/joining-mainnet/join-as-a-validator-using-statesync.md [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+order: 6
+-->
+
+# How to Join as a Validator using State Sync
+
+In this section, we'll explain the requirements and basics for running your own BandChain validator node using [State Sync](https://blog.cosmos.network/cosmos-sdk-state-sync-guide-99e4cf43be2f).
+
+## Warning
+
+You have to have at least 32 GB of RAM to using State Sync
```
Comment (RogerKSI) [inline docs/joining-mainnet/join-as-a-validator-using-statesync.md]: should be "to use"?

File: docs/joining-mainnet/join-as-a-validator.md [hunk 4]
```diff
@@ -90,9 +92,55 @@ bandd keys add $WALLET_NAME
 This can be done by editing `seeds` or `persistent_peers` property in `$HOME/.band/config/config.toml`.
 Please see [here](https://github.com/bandprotocol/launch/tree/master/laozi-mainnet) for the list of seeds and peers.
 
-### Step 1.5: Create BandChain service
+```bash
+# List of seeds and persistent peers you want to add
+export SEEDS="<SEED>,<SEED>,..." 
+# e.g. SEEDS="8d42bdcb6cced03e0b67fa3957e4e9c8fd89015a@34.87.86.195:26656,543e0cab9c3016a0e99775443a17bcf163038912@34.150.156.78:26656"
```
Comment (RogerKSI) [inline docs/joining-mainnet/join-as-a-validator.md]: I think e.g. comment should come before the command but up to you krub.

File: docs/joining-mainnet/join-as-a-validator-using-statesync.md [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+sed -E -i \
+  "s/persistent_peers = \".*\"/persistent_peers = \"${PERSISTENT_PEERS}\"/" \
+  $HOME/.band/config/config.toml
+```
+
+### Step 1.5: Setup State Sync config
+
+```bash
+# Get trust height and trust hash
+
```
Comment (RogerKSI) [inline docs/joining-mainnet/join-as-a-validator-using-statesync.md]: no empty line.

File: docs/joining-mainnet/join-as-a-validator-using-statesync.md [hunk 6]
```diff
@@ -0,0 +1,385 @@
+<!--
```
Comment (RogerKSI) [inline docs/joining-mainnet/join-as-a-validator-using-statesync.md]: Will the filename be the section name? should we align it with testnet file?

File: docs/joining-mainnet/join-as-a-validator-using-statesync.md [hunk 7]
```diff
@@ -0,0 +1,385 @@
...
+```
+
+### Step 1.5: Setup State Sync config
+
+```bash
+# Get trust height and trust hash
+
+LATEST_HEIGHT=$(curl -s http://rpc.laozi1.bandchain.org/block | jq -r .result.block.header.height);
+TRUST_HEIGHT=$(($LATEST_HEIGHT-45000))
+TRUST_HASH=$(curl -s "http://rpc.laozi1.bandchain.org/block?height=$TRUST_HEIGHT" | jq -r .result.block_id.hash)
```
Comment (RogerKSI) [inline docs/joining-mainnet/join-as-a-validator-using-statesync.md]: sry, please change it to laozi4.
- it's https.
- it's full rest.

File: docs/joining-mainnet/join-as-a-validator-using-statesync.md [hunk 8]
```diff
@@ -0,0 +1,385 @@
...
+
+```bash
+# Enable State Sync
+sed -i \
+    '/\[statesync\]/,+34 s/enable = false/enable = true/' \
+    $HOME/.band/config/config.toml
+
+# Set RPC Endpoint for State Sync
+sed -E -i \
+    "/\[statesync\]/,+34 s/rpc_servers = \".*\"/rpc_servers = \"http\:\/\/rpc.laozi1.bandchain.org\:80,http\:\/\/rpc.laozi2.bandchain.org\:80,https\:\/\/rpc.laozi3.bandchain.org\:80,https\:\/\/rpc.laozi4.bandchain.org\:80\"/" \
```
Comment (RogerKSI) [inline docs/joining-mainnet/join-as-a-validator-using-statesync.md]: for 3,4, port should be 443.

File: docs/joining-mainnet/join-as-a-validator-using-statesync.md [hunk 9]
```diff
@@ -0,0 +1,385 @@
...
+We do recommend to run bandchain node as a daemon, which can be setup using `systemctl`. Run the following command to create a new daemon for `cosmovisor` that runs `bandd` (This script work on non-root user).
+
+```bash
+# Write bandd service file to /etc/systemd/system/bandd.service
+export USERNAME=$(whoami)
+sudo -E bash -c 'cat << EOF > /etc/systemd/system/bandd.service
+[Unit]
+Description=BandChain Node Daemon
+After=network-online.target
+[Service]
```
Comment (RogerKSI) [inline docs/joining-mainnet/join-as-a-validator-using-statesync.md]: add empty line between section (unit<->service, service<->install) to align with yoda service.

File: docs/joining-mainnet/join-as-a-validator-using-statesync.md [hunk 10]
```diff
@@ -0,0 +1,385 @@
...
+The following applications are required to build and run the BandChain node.
+
+- make, gcc, g++ (can be obtained from `build-essential` package on linux)
+- wget, curl for downloading files
+
+```bash
+# install required tools
+sudo apt-get update && \
+sudo apt-get upgrade -y && \
+sudo apt-get install -y build-essential curl wget
```
Comment (RogerKSI) [inline docs/joining-mainnet/join-as-a-validator-using-statesync.md]: install jq as well.

**Review by RogerKSI — CHANGES_REQUESTED** [review]

File: docs/joining-mainnet/join-as-a-validator-using-statesync.md (hunk 3 above)
Comment (colmazia) [reply docs/joining-mainnet/join-as-a-validator-using-statesync.md]: oh don't deserve G4

File: docs/joining-mainnet/join-as-a-validator-using-statesync.md (hunk 10 above)
Comment (colmazia) [reply docs/joining-mainnet/join-as-a-validator-using-statesync.md]: wow nice catch

File: docs/joining-testnet/join-as-a-testnet-validator.md [hunk 11]
```diff
@@ -0,0 +1,421 @@
+<!--
+order: 7
+-->
+
+# How to Join as a Validator
+
+In this section, we'll explain the requirements and basics for running your own BandChain testnet#6 validator node and syncing from the starting block. If you want to join using State Sync or Quick Sync, please follow next documents.
```
Comment (RogerKSI) [inline docs/joining-testnet/join-as-a-testnet-validator.md]: no quick sync for testnet.
last comment from me. :)

**Review by RogerKSI — CHANGES_REQUESTED** [review]

**Review by RogerKSI — APPROVED** [review]

**Pull request merged**